config/config_test.go

```diff
@@ -549,6 +552,92 @@ func TestGetSubnetConfigsFromFlags(t *testing.T) {
 	}
 }
 
+func TestGetStakingSigner(t *testing.T) {
+	testKey := "HLimS3vRibTMk9lZD4b+Z+GLuSBShvgbsu0WTLt2Kd4="
+	testKeyPath1 := filepath.Join(t.TempDir(), ".avalanchego/staking/signer.key")
+	testKeyPath2 := strings.Replace(testKeyPath1, "001", "002", 1) // Anticipate the new temp dir that will be created
```

Comment:
Is this safe to do? I think `t.TempDir` does clean up the dirs (so you can expect to start at 001), but if you hit power. loss, panic, a sig kill, or any other unexpected failure, this string replacement is brittle and can lead to flaky unit tests. It's also not documented as a guarantee on the interface for `TempDir`. Is it possible for us to get rid of this string matching?

Reply:
I think `t.TempDir` creates an entirely new parent directory per test run, so should be unaffected by power-loss, panic etc, but I did find a way to remove the string replacement

Reply:
In this case I'm fairly certain that `testing` will create an entirely new parent directory for the subsequent test run